src/Enum.php

```diff
@@ -172,6 +182,18 @@ public static function search($value)
         return \array_search($value, static::toArray(), true);
     }
 
+    /**
+     * Returns a title cased label based on a key
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    public static function getLabelForValue($value)
+    {
+        return static::keyToTitleCase(static::search($value));
```

Comment:
keep in mind, ::search() might return false when the key doesn't exist so this might end up passing false to `keyToTitleCase` and produce unexpected results/errors

Reply:
might want evaluate the return of ::search() and adjust to throw `\UnexpectedValueException` when the value is not  a valid one for the enum